### ISSUE TITLE
custom dataset

### ISSUE THREAD

**Issue by grisaiakaziki** [yashkant/spad#6]
Excellent work!
I have a question: if I need to use my own dataset, how should I process it? Also, does depth have a significant impact on performance? Thank you.

**Comment by grisaiakaziki** [comment]
How should I process the COLMAP dataset?

**Comment by yashkant** [comment]
thanks for checking out spad! 


> using own dataset

if you wish to train spad on custom data, you can use the dataloader from [zero123](https://github.com/cvlab-columbia/zero123/blob/main/zero123/ldm/data/simple.py#L208), which has the same coordinate system as spad. 

for each item in your dataset, can easily compute plucker coordinates and epipolar mask using this [function](https://github.com/yashkant/spad/blob/main/spad/geometry.py#L324) from spherical coordinates. these will be used during in the forward pass. 

> colmap dataset

you will have to preprocess colmap dataset by centering the points to origin and rescaling the scene to fixed bounds (say -1 to 1). also, you will have to convert them to blender/pytorch coordinate systems (see [here](https://miro.medium.com/v2/resize:fit:1400/1*neELJFLJMaO6ARdGVrqTwg.png) and [here](https://stackoverflow.com/questions/64977993/applying-opencv-pose-estimation-to-blender-camera)).
